core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceService.scala

```diff
@@ -141,22 +141,23 @@ class MultiCloudWorkspaceService(override val ctx: RawlsRequestContext,
         s"V2 Starting deletion of workspace [workspaceId=${workspace.workspaceId}, name=${workspaceName.name}, billingProject=${workspace.namespace}, user=${ctx.userInfo.userSubjectId.value}]"
       )
 
-      _ = workspace.state match {
-        case WorkspaceState.Ready | WorkspaceState.CreateFailed | WorkspaceState.DeleteFailed |
-            WorkspaceState.UpdateFailed =>
+      _ =
+        if (workspace.state.isDeletable) {
           true
-        case WorkspaceState.Deleting =>
-          throw new RawlsExceptionWithErrorReport(
-            ErrorReport(StatusCodes.Conflict, "Workspace is already being deleted.")
-          )
-        case _ =>
-          logger.error(
-            s"Unable to delete workspace [id = ${workspace.workspaceId}, current_state = ${workspace.state}]"
-          )
-          throw new RawlsExceptionWithErrorReport(
-            ErrorReport(StatusCodes.BadRequest, s"Unable to delete workspace")
-          )
-      }
+        } else {
+          if (workspace.state == WorkspaceState.Deleting) {
```

Comment:
Note that there is some test coverage of this code in https://github.com/broadinstitute/rawls/blob/9452d13027d83a2e02f4ad76715086d4d9a8016d/core/src/test/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceServiceSpec.scala#L1498

I decided not to add a test case specifically for CloneFailed, but to instead just test the new `isDeletable` property of WorkspaceState.